docs/architecture.rst

```diff
@@ -1,66 +1,172 @@
+:orphan:
```

Comment:
This was added because the documentation build was providing a warning `WARNING: Field list ends without a blank line; unexpected unindent. [docutils]`.

I think this is because I added a cross-reference target to the document such that it could be referenced elsewhere.
![image](https://github.com/user-attachments/assets/17de93b7-fa6b-4fec-8206-8405fad4ce29)

Reply:
I was able to solve that locally by tweaking a little bit:

```rst
A high-level overview of the different modules in ``tqec`` is available in 

.. toctree::
   :maxdepth: 1

   architecture
```

I think this is a reasonable compromise. Also, I prefer having the link also on the left bar containing links, I personally find that easier to navigate.